src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -133,40 +149,50 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     if (!queueEntry.IsAbandoned && !queueEntry.IsCompleted)
                         await queueEntry.AbandonAsync().AnyContext();
                 }
+            }, new OnMessageOptions {
+                AutoComplete = false
             });
         }
 
         public override async Task<IQueueEntry<T>> DequeueAsync(TimeSpan? timeout = null) {
+            await EnsureQueueCreatedAsync().AnyContext();
```

Comment:
And one last thing before I have to go: there's another race condition here (and with all calls to `EnsureQueueCreatedAsync`). If another thread calls `DeleteQueueAsync` immediately after this call returns, the call to `_queueClient.ReceiveAsync` will throw.

I tried to solve this in my previous PR by having `EnsureQueueCreatedAsync` return a reference to the queue, so it at least wouldn't throw `NullReferenceException`, but if the queue is deleted then you'll of course get a different exception related to that.